homeassistant/components/enigma2/media_player.py

```diff
@@ -0,0 +1,196 @@
+    name = config[CONF_NAME]
+
+    # Generate a configuration for the Enigma2 library
+    remote_config = {
+        'name': 'HomeAssistant',
+        'description': config[CONF_NAME],
+        'port': config[CONF_PORT],
+        'host': config[CONF_HOST],
+        'timeout': DEFAULT_TIMEOUT,
+    }
+
+    add_devices([Enigma2Device(name, remote_config)], True)
+
+
+class Enigma2Device(MediaPlayerDevice):
+    """Representation of an Enigma2 box."""
+
+    def __init__(self, name, config):
+        """Initialize the Enigma2 device."""
+        import openwebif.api
+        self._state = None
+        self._name = name
+        self.e2_box = openwebif.api.CreateDevice(config['host'],
+                                                 port=config['port'])
+
+        self.volume = 10
+        self.current_service_channel_name = None
+        self.current_programme_name = None
+        self.current_service_ref = None
+        self.muted = False
+        self.picon_url = None
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def supported_features(self):
+        """Flag of media commands that are supported."""
+        return SUPPORTED_ENIGMA2
+
+    def turn_off(self):
+        """Turn off media player."""
+        if self.state == STATE_ON:
+            self.e2_box.toggle_standby()
+
+    def turn_on(self):
+        """Turn the media player on."""
+        if self.state == STATE_OFF:
+            self.e2_box.toggle_standby()
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        if self.current_service_ref.startswith('1:0:0'):
+            return "[Recording Playback] - {}".format(
+                self.current_programme_name)
+        if self.current_service_channel_name:
+            return "{} - {}".format(self.current_programme_name,
+                                    self.current_service_channel_name)
+
+        return self.current_programme_name
+
+    @property
+    def media_channel(self):
+        """Channel of current playing media."""
+        return self.current_service_channel_name
+
+    @property
+    def media_content_id(self):
+        """Service Ref of current playing media."""
+        return self.current_service_ref
+
+    @property
+    def media_content_type(self):
+        """Type of video currently playing."""
+        return MEDIA_TYPE_TVSHOW
+
+    @property
+    def is_volume_muted(self):
+        """Boolean if volume is currently muted."""
+        return self.muted
+
+    @property
+    def media_image_url(self):
+        """Picon url for the channel."""
+        return self.picon_url
+
+    def set_volume_level(self, volume):
+        """Set volume level, range 0..1."""
+        self.e2_box.set_volume(int(volume * 100))
+        self.volume = volume * 100
```

Comment:
Don't do this. Home Assistant will call your update method after calling this method.